agent/consul/leader_peering_test.go

```diff
@@ -1036,3 +1036,89 @@ func TestLeader_PeeringMetrics_emitPeeringMetrics(t *testing.T) {
 		require.Equal(r, float32(2), metric2.Value) // for d, e services
 	})
 }
+
```

Comment:
These tests take 7s each because they wait for nothing to happen and if peering was enabled it takes about 7s for something to happen. It is important that the behaviour is tested though so probably worth the time.